include/boost/json/monotonic_resource.hpp

```diff
@@ -72,36 +98,126 @@ class monotonic_resource final
+
+        @par Complexity
+
+        Linear in the number of allocated blocks.
+    */
     BOOST_JSON_DECL
-    ~monotonic_resource();
+    ~monotonic_resource() noexcept;
+
+    /** Constructor.
+
+        Constructs a `monotonic_resource` whose initial size
+        is `initial_size` rounded up to the nearest power of two;
+        if `initial_size` is less than 1024, the initial size will
+        be 1024.
 
+        @note This constructor does not cause memory to be allocated.
+
+        @param initial_size The initial size of the resource.
+        The default argument for this parameter is 1024.
+    */
     BOOST_JSON_DECL
     monotonic_resource(
         std::size_t initial_size = 1024) noexcept;
 
+    /** Constructor.
+
+        Constructs a `monotonic_resource` whose initial buffer is
+        the `buffer_size` bytes of memory pointed to by `buffer`.
+        The initial size is `buffer_size` rounded up to the nearest
+        power of two; if `buffer_size` is less than 1024, the initial
+        size will be 1024.
+
+        @par Precondition
+
+        `{buffer, buffer + buffer_size)` is a valid range.
+
+        @note This constructor does not cause memory to be allocated,
+        and does not take ownership of the initial buffer.
+
+        @param buffer A pointer to the initial buffer for the
+        resource to use.
+        
+        @param buffer_size The size of the initial buffer.
+    */
     BOOST_JSON_DECL
     monotonic_resource(
         void* buffer,
         std::size_t buffer_size) noexcept;
 
+    /** Allocate memory.
+
+        Allocates a region of memory of size `n`
+        that is suitably aligned for `align`.
+        
+        If the `monotonic_resource` has an initial buffer
+        and its initial buffer has sufficent space to meet
+        the request, the memory will be allocated within
+        that buffer.
+
+        Otherwise, if the current dynamically allocated block
+        has sufficent room to meet the request, the memory will
+        be allocated within that block. If there is no such block,
+        or if the current block cannot meet the request, a new block
+        will be allocated. The size of the new block will be twice
+        as large as that of the current block, or `n` rounded up to
+        the nearest power of two if `n` exceeds the size of the current
+        block. The memory is then allocated within the new block.
+
+        @par Exception Safety
+
+        Strong guarantee.
+
+        @note Once a new block is dynamically allocated,
+        memory will never be allocated in any previous blocks
+        even if they have sufficent room to fulfil the request.
+
+        @return A pointer to memory that is of size `n` and
+        suitably aligned for `align`.
+
+        @param n The number of bytes to allocate.
+
+        @param align The desired alignment of the memory.
+
+        @throw std::bad_alloc Failure to allocate a new block.
+    */
```

Comment:
This is not going to work out well, we need to discuss it

Reply:
The way the exception is specified? Or is it because it doesn't directly throw the exception?

I can just add a not saying that it calls `::operator new` and that might throw.

Reply:
Because of how it is derived from memory_resource - we talked about this already.

Reply:
Ah, we resolved that already then.